src/core/Table/hooks/useColumnVisibility.tsx

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { ColumnInstance, HeaderProps, Hooks } from 'react-table';
+import { Checkbox } from '../..';
+import SvgColumnManager from '@itwin/itwinui-icons-react/cjs/icons/ColumnManager';
+import { DropdownMenu } from '../../DropdownMenu';
+import { IconButton } from '../../Buttons/IconButton';
+import { MenuItem } from '../../Menu';
+
+export const COLUMN_MANAGER_ID = 'iui-table-column-manager';
+
+export const useColumnVisibility = <T extends Record<string, unknown>>(
+  hasColumnManager: boolean,
+) => (hooks: Hooks<T>) => {
+  if (!hasColumnManager) {
+    return;
+  }
+
+  hooks.allColumns.push((columns: ColumnInstance<T>[]) => [
+    ...columns,
+    {
+      id: COLUMN_MANAGER_ID,
```

Comment:
After #599 is merged, you might want to move this and export as separate column.
Then it should be possible to check if such column already exist and add menu to that header OR just add whole new column

Reply:
updated